saleor/order/models.py

```diff
@@ -500,3 +500,10 @@ class Meta:
 
     def __repr__(self):
         return f"{self.__class__.__name__}(type={self.type!r}, user={self.user!r})"
+
+
+class Invoice(models.Model):
```

Comment:
I don't know what are the plans for Invoicing, but please be aware that depending on countries, the voice identifier (ID) can have different regulations. In France we recommend using a prefix, e.g. `FR000001`, `SHOP000001`.

Only important rule for France is that they have to be continuous (growing) numbers, and must have no gaps, e.g. `FR202003041` is illegal.

Anyway apart the examples, to come back to the point, maybe we would want to allow meta-data and extensions for invoices so people can adapt if they have special or odd needs?

Reply:
Yes, we will have metadata for invoices but that's not in the scope of this task. To ensure correct IDs - it will be implemented at the plugin level, so you can have different plugins to handle different formats.